packages/webpack/src/utils/config.ts

```diff
@@ -68,5 +68,5 @@ export function fileName (ctx: WebpackConfigContext, key: string) {
 
 export function getWebpackConfig (ctx: WebpackConfigContext): Configuration {
   // Clone deep avoid leaking config between Client and Server
-  return cloneDeep(ctx.config)
+  return defu({}, ctx.config)
```

Comment:
Please notice that defu only clones on merge not always. For safer alternative, i would suggest clona (we use it for nitro config too)